moveit_simple/test/kuka_kr210.cpp

```diff
@@ -146,6 +146,25 @@ TEST_F(UserRobotTest, add_trajectory)
   EXPECT_THROW(robot->execute("bad_traj"), std::invalid_argument);
 }
 
+TEST_F(UserRobotTest, validate_trajectory)
+{
+  ROS_INFO_STREAM("Testing trajectory validation");
+  const std::string TRAJECTORY_NAME("val_traj");
+  const moveit_simple::InterpolationType joint = moveit_simple::interpolation_type::JOINT;
+
+  // try valid trajectory
+  EXPECT_NO_THROW(robot->addTrajPoint(TRAJECTORY_NAME, "home",      0.5));
+  EXPECT_NO_THROW(robot->addTrajPoint(TRAJECTORY_NAME, "waypoint1", 1.0, joint, 5));
+  EXPECT_NO_THROW(robot->execute(TRAJECTORY_NAME));
```

Comment:
Suggestion: Add a test that calls `trajectory_processing::validateTrajectory` directly?